### PR TITLE
[DON-891] add annotated string to calendar day

### PR DESCRIPTION
**Pull request by peterInTown**
Add an annotated string to the calendar day text so that consumers can identify the first active cells for a smoke test
https://github.skyscannertools.net/apps-tribe/skyscanner-app/pull/38479

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt [hunk 1]
```diff
@@ -82,6 +84,7 @@ internal fun BpkCalendarDayCell(
                 interactionSource = remember { MutableInteractionSource() },
             )
             .semantics {
+                text = AnnotatedString(if (inactive) "inactive" else "active")
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:87]: what happens with screenreaders here? does it announce this?

Comment (peterInTown) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:87]: I have double check it and screenreader doesn't announce it

Comment (henrik-sky) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:87]: Hmm, I'm seeing/hearing it announced
Screenshot_20241129_091246
Screenshot_20241129_091224

Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:87]: ok we definitely can't do this then

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt [hunk 2]
```diff
@@ -81,7 +85,9 @@ internal fun BpkCalendarDayCell(
                 onClickLabel = model.onClickLabel,
                 interactionSource = remember { MutableInteractionSource() },
             )
+            .testTag(if (inactive) "inactive" else "active")
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:88]: Is it just inactive/active that's needed as a test tag here or some more info around the date?

Comment (peterInTown) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/calendar/internal/BpkCalendarDayCell.kt:88]: Screenshot 2024-11-27 at 16 46 32


For active and inactive is mainly for detecting selectable and not selectable days. For the smoke test purpose inactive/active satisfied the need for now. If consumer need more information about the day, each day cell already contains contentdescription such as "Tuesday, 11 November 2023

**Review by marianeum — APPROVED** [review]

**Pull request merged**